graylog2-server/src/test/java/org/graylog/plugins/views/search/searchtypes/pivot/buckets/TimeUnitIntervalTest.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.views.search.searchtypes.pivot.buckets;
+
+import org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramInterval;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+import org.graylog2.plugin.indexer.searches.timeranges.RelativeRange;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
```

Comment:
Nice and clean tests! I think they could be further improved by using parameterized tests. That would make them even more compact and facilitate adding more cases. Totally optional though. The tests are fine as they are now.

Reply:
I was also thinking about that, but I hesitated doing it, because afaik it requires all methods of a test class to be parameterized, so the test class would have to be split up into one for the validation tests and one for the valid inputs. I can still do it though, I just preferred to have a single test class right now, because it makes it easier to spot corner cases which are not covered.

Reply:
Well no, I can just wrap those in an inner class. Will do!

Reply:
Right. Forgot that we're on JUnit 4. JUnit 5 has much better support for parameterized tests. For the short term we should consider using https://github.com/Pragmatists/JUnitParams then.